### PR TITLE
Total flights stats for OSD (configurator part)

### PR DESCRIPTION
**Pull request by limonspb**
Adding total flights count from persistent stats to OSD.
Configurator part
Feature request: https://github.com/betaflight/betaflight/issues/10215

Pull request for firmware part: https://github.com/betaflight/betaflight/pull/10300

### CODE REVIEW

File: src/js/tabs/osd.js [hunk 1]
```diff
@@ -1172,7 +1172,14 @@ OSD.loadDisplayFields = function() {
...
         },
-
+        PERSISTENT_TOTAL_FLIGHTS: {
+            name: 'OSD_PERSISTENT_TOTAL_FLIGHTS',
+            text: 'osdTextTotalFlights',
+            desc: 'osdDescTotalFlights',
+            defaultPosition: -1,
+            positionable: true,
+            preview: "9876",
+        },
```
Comment (haslinghuis) [inline src/js/tabs/osd.js:1182]: Where is draw_order being used for?

**Review by haslinghuis — DISMISSED** [review]

File: src/js/tabs/osd.js (hunk 1 above)
Comment (McGiverGim) [reply src/js/tabs/osd.js:1182]: Is like z-index for osd elements. It must have a value accordingly with the order in the firmware.

Comment (haslinghuis) [reply src/js/tabs/osd.js:1182]: Limon, I think `draw_order` has to be `460` here, could you please add.

Comment (limonspb) [reply src/js/tabs/osd.js:1182]: I saw not all elements has it, so should be a default value, right?

Comment (McGiverGim) [reply src/js/tabs/osd.js:1182]: All the osd elements have this. If someone is missing we need to add them. This makes the preview more realistic when a user mounts one element over other.

File: locales/en/messages.json [hunk 2]
```diff
@@ -4875,7 +4875,14 @@
     "osdDescElementEfficiency": {
         "message": "Instantaneous battery consumption in mAh/distance.  (Requires valid GPS fix)"
     },
-    
+    "osdTextTotalFlights": {
+        "message": "Total flights",
+        "description": "One of the elements of the OSD"
+    },
+    "osdDescTotalFlights": {
+        "message": "Estimated total flights"
```
Comment (mikeller) [inline locales/en/messages.json:4883]: Not sure 'estimated' is the right word here - we actually count them and do not use some estimation technique. Would 'approximate' be better, to account for the fact that this is not necessarily foolproof? Probably also better make it 'Approximate total number of flights'.

File: src/js/tabs/osd.js (hunk 1 above)
Comment (haslinghuis) [reply src/js/tabs/osd.js:1182]: So far I've spotted these elements not having a `draw_order` value.
They are in the same function `OSD.loadDisplayFields = function() {};` but not necessary relate to the OSD elements in firmware?
```
TIMER
CPU_LOAD
VOLTAGE_WARNING
ARMED_TIME
ARMED
```
Not all occurrences using the value are placed in order. Only an observation and not part of this PR.

Comment (haslinghuis) [reply src/js/tabs/osd.js:1182]: For this PR i would suggest a value of `460`, but it's just my assumption.

Comment (limonspb) [reply src/js/tabs/osd.js:1182]: @haslinghuis  im not completely understanding how should i select the value. Could you please explain? Thanks

Comment (haslinghuis) [reply src/js/tabs/osd.js:1182]: If have no clue either - values seems arbitrary to me. Let's wait for @McGiverGim

Comment (haslinghuis) [reply src/js/tabs/osd.js:1182]: To check try the line (after line 1180):
```
draw_order: 460, 
```
and see if there is a difference?
But I guess this was obvious.

Comment (limonspb) [reply src/js/tabs/osd.js:1183]: yeah they are kinda following in the order as I can see. Thanks!

Comment (McGiverGim) [reply src/js/tabs/osd.js:1183]: The OSD elements are drawn in certain order in firmware:

https://github.com/betaflight/betaflight/blob/60c008b20bd025eadf0d7b7eb6a35c5e9f615756/src/main/osd/osd_elements.c#L1556-L1625

The OSD elements in the preview must be drawn in the same order. In your firmware PR you are adding it after `OSD_EFFICIENCY`, that has `draw_order: 455,` so you need to add some numbers to this. We usually add 5 to this number, maybe in the future we want to add some element between them. So 460 is a good number.

Usually this order does not matter, but some elements need to be over other. The crosshair must be over the horizon artificial, the warnings over all of the other osd elements, etc. Some users mount some elements over others to occupy less space. This gives them an accurate preview.

The elements that you have found and don't have preview are legacy elements that do not exists in the current firmwares.

**Review by McGiverGim — APPROVED** [review]

**Review by mikeller — APPROVED** [review]

**Pull request merged**